webapp/src/ts/services/enketo.service.ts

```diff
@@ -686,16 +707,25 @@ export class EnketoService {
   }
 
   private create(formInternalId) {
-    return this.getUserContact().then((contact) => {
-      return {
-        form: formInternalId,
-        type: 'data_record',
-        content_type: 'xml',
-        reported_date: Date.now(),
-        contact: this.extractLineageService.extract(contact),
-        from: contact && contact.phone
-      };
-    });
+    return this
+      .getUserContact()
+      .then(contact => {
+        const doc:any = {
+          form: formInternalId,
+          type: 'data_record',
+          content_type: 'xml',
+          reported_date: Date.now(),
+          contact: this.extractLineageService.extract(contact),
+          from: contact && contact.phone
+        };
+
+        if (this.isTrainingForm(formInternalId)) {
+          doc._id = `${this.TRAINING_PREFIX}${this.sessionService.userCtx()?.name}:${uuid()}`;
```

Comment:
It'd be easier to read if this were done with a join, eg:

```suggestion
          doc._id = [ this.TRAINING_PREFIX, this.sessionService.userCtx()?.name, uuid() ].join(':');
```

This would require also changing the TRAINING_PREFIX to not have the `:` included but I think that's an improvement.

I would probably also have a `const` for the username so it's not all in one line, eg:

```suggestion
          const username = this.sessionService.userCtx()?.name;
          doc._id = [ this.TRAINING_PREFIX, username, uuid() ].join(':');
```